pxr/imaging/plugin/hdRpr/rprApi.cpp

```diff
@@ -1466,69 +1473,96 @@ class HdRprApiImpl {
             return nullptr;
         }
 
-        auto cubeMesh = CreateCubeMesh(1.0f, 1.0f, 1.0f);
-        if (!cubeMesh) {
-            return nullptr;
-        }
-
         LockGuard rprLock(m_rprContext->GetMutex());
 
         auto rprApiVolume = new HdRprApiVolume;
 
-        rprApiVolume->cubeMeshMaterial.reset(HdRprApiRawMaterial::Create(m_rprContext.get(), RPR_MATERIAL_NODE_TRANSPARENT, {
-            {RPR_MATERIAL_INPUT_COLOR, GfVec4f(1.0f)}
-        }));
-        rprApiVolume->cubeMesh.reset(cubeMesh);
+        rpr::Status status;
+
+        rprApiVolume->base_mesh.reset(CreateVoidMesh());
+        rprApiVolume->volumeShader.reset(m_rprContext->CreateMaterialNode(RPR_MATERIAL_NODE_VOLUME, &status));
+        rprApiVolume->densityGridShader.reset(m_rprContext->CreateMaterialNode(RPR_MATERIAL_NODE_GRID_SAMPLER, &status));
+
+        //Northstar does not support density lookup, so we need to compute values
+        std::vector<float> computedDensityValues;
+        computedDensityValues.resize(densityValues.size());
+
+        for (int idx = 0; idx < densityValues.size(); idx++) {
+            if (densityValues[idx] < 0) {
+                computedDensityValues[idx] = densityLUT[0][0];
+                continue;
+            }
+            if (densityValues[idx] >= 1) {
+                computedDensityValues[idx] = densityLUT.back()[0];
+                continue;
+            }
+            size_t lookupIndex = floor(densityValues[idx] * (densityLUT.size() - 1));
+            
+            //linear interpolation
+            float firstValue = densityLUT[lookupIndex][0];
```

Comment:
Is that guaranteed that lookupIndex is within the vector bounds?

Reply:
Yes because we check index in lines 1490-1498